src/sqlfluff/utils/reflow/elements.py

```diff
@@ -0,0 +1,383 @@
+        prev_block: Optional[ReflowBlock] = None,
+        next_block: Optional[ReflowBlock] = None,
+        fixes: Optional[List[LintFix]] = None,
+        strip_newlines: bool = False,
+    ) -> Tuple[List[LintFix], "ReflowPoint"]:
+        """Respace a point based on given constraints.
+
+        NB: This effectively includes trailing whitespace fixes.
+
+        Deletion and edit fixes are generated immediately, but creations
+        are paused to the end and done in bulk so as not to generate conflicts.
+
+        Note that the `strip_newlines` functionality exists here as a slight
+        exception to pure respacing, but as a very simple case of positioning
+        line breaks. The default operation of `respace` does not enable it
+        however it exists as a convenience for rules which wish to use it.
+        """
+        new_fixes = []
+        last_whitespace: List[RawSegment] = []
+        # The buffer is used to create the new reflow point to return
+        segment_buffer = list(self.segments)
+        edited = False
+        pre_constraint = prev_block.spacing_after if prev_block else "single"
+        post_constraint = next_block.spacing_before if next_block else "single"
+
+        # Work out the common parent segment and depth
+        if prev_block and next_block:
+            common = prev_block.depth_info.common_with(next_block.depth_info)
+            # Just check the most immediate parent for now for speed.
+            # TODO: Review whether this is enough.
+            # NOTE: spacing configs will be available on both sides if they're common
+            # so it doesn't matter whether we get it from prev_block or next_block.
+            within_constraint = prev_block.stack_spacing_configs.get(common[-1], None)
+            if not within_constraint:
+                pass
+            elif within_constraint in ("touch", "inline"):
+                # NOTE: inline is actually a more extreme version of "touch".
+                # i.e. inline, implies no spaces between either.
+                if within_constraint == "inline":
+                    # If they are then strip newlines.
+                    strip_newlines = True
+                # If segments are expected to be touch within. Then modify
+                # constraints accordingly.
+                # NOTE: We don't override if it's already "any"
+                if pre_constraint != "any":
+                    pre_constraint = "touch"
+                if post_constraint != "any":
+                    post_constraint = "touch"
+            else:  # pragma: no cover
+                idx = prev_block.depth_info.stack_hashes.index(common[-1])
+                raise NotImplementedError(
+                    f"Unexpected within constraint: {within_constraint} for "
+                    f"{prev_block.depth_info.stack_class_types[idx]}"
+                )
+
+        reflow_logger.debug("Respacing: %s", self)
+        for idx, seg in enumerate(self.segments):
```

Comment:
Nitpick: It doesn't appear that `idx` is used, thus the `enumerate()` could be removed.

Reply:
Good catch!